src/utils/limits.js

```diff
@@ -0,0 +1,49 @@
+export const MAX_STRUCTS = 100_000
+export const MAX_UPDATES = 100_000
+export const MAX_GC_LENGTH = 100_000
+export const MAX_SKIP_LENGTH = 100_000
```

Comment:
These limits are totally arbitrary. I honestly don't know what number would make sense here. 

Another approach I thought about was to increment a loop counter, and use the same limit for structs and updates. I don't think this would work for GC and skip item lengths, though.